api/inc/context_exports.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2016, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef __UVISOR_CONTEX_EXPORTS_H__
+#define __UVISOR_CONTEX_EXPORTS_H__
+
+/** Maximum number of nested context switches. The same state stack is kept for
+ * all kinds of context switches, which includes secure gateways, interrupts,
+ * and task switching. */
+#define UVISOR_CONTEXT_MAX_DEPTH 16
```

Comment:
Not strictly needed by user code with our current use-cases, but we use it in tests.

I can imagine though that it could be use to ensure that a recursive function doesn't go too deep before hitting a `HALT_ERROR`.